massa-consensus-worker/src/tests/inter_cycle_batch_finalization.rs

```diff
@@ -0,0 +1,189 @@
+/// * ensure that when B is added:
+///   * C2 becomes stale
+///   * a batch of blocks from B1 becomes final so that this batch spans across the limit of a cycle
+///
+/// If the lookup is badly implemented, B's verification step will fail
+/// because Consensus and PoS are desynchronized in their final block/cycle knowledge.
+/// If the the lookup is implemented correctly, B will be propagated.
+#[tokio::test]
+#[serial]
+async fn test_inter_cycle_batch_finalization() {
+    let t0: MassaTime = 1000.into();
+    let staking_key = generate_random_private_key();
+    let creator_public_key = derive_public_key(&staking_key);
+    let creator_addr = Address::from_public_key(&creator_public_key);
+    let roll_price = Amount::from_str("42").unwrap();
+    let initial_ledger = vec![(
+        creator_addr,
+        LedgerData {
+            balance: roll_price, // allows the address to buy 1 roll
+        },
+    )]
+    .into_iter()
+    .collect();
+    let warmup_time: MassaTime = 1000.into();
+    let margin_time: MassaTime = 300.into();
+    let cfg = ConsensusConfig {
+        periods_per_cycle: 4,
+        delta_f0: 2,
+        thread_count: 1,
+        endorsement_count: 2,
+        roll_price,
+        t0,
+        future_block_processing_max_periods: 50,
+        genesis_timestamp: MassaTime::now().unwrap().saturating_add(warmup_time),
+        ..ConsensusConfig::default_with_staking_keys_and_ledger(&vec![staking_key], &initial_ledger)
+    };
+
+    consensus_without_pool_test(
+        cfg.clone(),
+        async move |mut protocol_controller, consensus_command_sender, consensus_event_receiver| {
+            // wait for consensus warmup time
+            tokio::time::sleep(warmup_time.to_duration()).await;
+
+            let genesis_blocks: Vec<BlockId> = consensus_command_sender
+                .get_block_graph_status(None, None)
+                .await
+                .expect("could not get block graph status")
+                .best_parents
+                .iter()
+                .map(|(b, _p)| *b)
+                .collect();
+
+            // Graph (1 thread):
+            //            cycle limit
+            //                v
+            // G - M - A1 - M - A2 - A3
+            //   \
+            //    B1 - M  - B2
+            //
+            // Where:
+            // G = genesis
+            // M = miss
+            // A1 = block buying 1 roll, 0 endorsements
+            // A2 = empty block, 0 endorsements
+            // A3 = block selling 1 roll, 2 endorsements
+            // B1, B2 = empty blocks of the alternative clique, 0 endorsements
+            //
+            // When A3 arrives:
+            // * B1 and B2 become stale
+            // * A1, A2 become final
+            // * A3 should be propagated
+
+            // Create, send and propagate B1
+            let (b1_id, b1_block, _) =
+                create_block(&cfg, Slot::new(1, 0), genesis_blocks.clone(), staking_key);
```

Comment:
Why not use create and serd block everywhere ?

Reply:
corrected everything